pkg/cloudproviders/aws/events/recorder.go

```diff
@@ -15,19 +15,52 @@ limitations under the License.
 package events
 
 import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/tools/record"
 )
 
-// Recorder is used to record events to the Kubernetes Event API
+type EventType byte
+
+const (
+	_ EventType = iota
+	NormalType
+	WarningType
+)
+
+func (e EventType) String() string {
+	switch e {
+	case NormalType:
+		return "Normal"
+	case WarningType:
+		return "Warning"
+	default:
+		return fmt.Sprintf("Unsupported EventType %d", e)
+	}
+}
+
+type Event interface {
+	InvolvedObject() runtime.Object
```

Comment:
Thoughts?
```suggestion
	For() runtime.Object
```

Reply:
I like `InvolvedObject` since it aligns with what `corev1.Event` calls the field

Reply:
Oh I missed that since the Event() API calls it object. SGMT.